stand/libsa/strcasecmp.c

```diff
@@ -39,34 +39,39 @@ static char sccsid[] = "@(#)strcasecmp.c	8.1 (Berkeley) 6/4/93";
 #endif /* LIBC_SCCS and not lint */
 
 int
-strcasecmp(s1, s2)
-	const char *s1, *s2;
+strcasecmp(const char *s1, const char *s2)
 {
-	const u_char
-			*us1 = (const u_char *)s1,
-			*us2 = (const u_char *)s2;
+	const u_char *us1 = (const u_char *)s1,
+	    *us2 = (const u_char *)s2;
 
-	while (tolower(*us1) == tolower(*us2++))
-		if (*us1++ == '\0')
+	u_char u1, u2;
+
+	while ((u1 = tolower(*us1)) == (u2 = tolower(*us2))) {
+		if (u1 == '\0') {
 			return (0);
-	return (tolower(*us1) - tolower(*--us2));
+		}
+		us1++, us2++;
+	}
+	return (u1 - u2);
```

Comment:
All of these changes should likely be in a separate pull request. It's not a simple cleanup, but completely rewrites this function. I'm honestly not sure that benefit from having the extra intermediaries, but it seems different enough to move to it's own pull request. The other cleanup could go together with the ANSI prototypes stuff, but this goes beyond a tweak.